packages/react-geo/src/components/MapView/__tests__/MapView.test.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { act, render, screen } from '@testing-library/react';
+import { Amplify, Auth } from 'aws-amplify';
+
+import '../../../../jest.setup';
+import { MapView } from '..';
+
+jest.mock('maplibre-gl-js-amplify', () => ({
+  AmplifyMapLibreRequest: jest.fn().mockImplementation(() => {
+    return {
+      amplifyTransformRequest: true,
+      transformRequest: (url: URL | string, resourceType: string) => {
+        if (
+          resourceType === 'Style' &&
+          typeof url === 'string' &&
+          !url.includes('://')
+        ) {
+          url = `https://maps.geo.us-east-2.amazonaws.com/maps/v0/maps/${url}/style-descriptor`;
+        }
+        if (new URL(url).hostname.endsWith('.amazonaws.com')) {
+          return {
+            url: 'https://maps.geo.us-east-2.amazonaws.com/maps/v0/maps/map5df169f7-staging/style-descriptor',
+          };
+        }
+      },
+    };
+  }),
+}));
+
+// Mock react-map-gl to check just that ReactMapGL component is rendered, and not what it actually renders.
+jest.mock(
+  'react-map-gl',
+  () =>
+    function ReactMapGlMock() {
+      return <div data-testid="react-map-gl-mock" />;
+    }
+);
+
+const configureSpy = jest.spyOn(Amplify, 'configure');
+const partialAmplifyConfig = {
+  geo: {
+    amazon_location_service: {
+      region: 'us-east-1',
+    },
+  },
+};
+
+const currentUserCredentialsSpy = jest.spyOn(Auth, 'currentUserCredentials');
+const currentUserCredentials = {
+  accessKeyId: '001122',
+  authenticated: true,
+  sessionToken: 'abcabc',
+  secretAccessKey: 'abcabc',
+  identityId: 'us-east-2:abcabc',
+};
+currentUserCredentialsSpy.mockImplementation(() =>
+  Promise.resolve(currentUserCredentials)
+);
+
+describe('Map component', () => {
+  it('should render', async () => {
+    configureSpy.mockReturnValue(partialAmplifyConfig);
+    render(<MapView />);
+    expect(await screen.findByTestId('react-map-gl-mock')).toBeInTheDocument();
```

Comment:
Removed `act` as it is not needed here.
https://github.com/aws-amplify/amplify-ui/pull/3314#discussion_r1132722342